### PR TITLE
Reintroduce buildinfo computation parallelism

### PR DESCRIPTION
**Pull request by ghale**
This supercedes #275 and fixes the issues reported there.  I've tested this with Gradle 4.10.3, 5.6.4, and 6.0 using the example project [here](https://github.com/jfrog/project-examples/blob/master/gradle-examples/gradle-example-publish/build.gradle).

I'm happy to discuss if any of the changes are unclear.

### CODE REVIEW

File: build-info-extractor-gradle/src/main/groovy/org/jfrog/gradle/plugin/artifactory/extractor/GradleBuildInfoExtractor.java [hunk 1]
```diff
@@ -324,11 +380,12 @@ public Artifact apply(GradleDeployDetails from) {
...
         List<Dependency> dependencies = newArrayList();
         for (Configuration configuration : configurationSet) {
-            if (configuration.getState() != Configuration.State.RESOLVED) {
+            DefaultConfiguration defaultConfiguration = (DefaultConfiguration) configuration;
+            if (defaultConfiguration.getResolvedState().compareTo(ConfigurationInternal.InternalState.GRAPH_RESOLVED) < 0) {
                 log.info("Artifacts for configuration '{}' were not all resolved, skipping", configuration.getName());
                 continue;
             }
-            ResolvedConfiguration resolvedConfiguration = getResolvedConfiguration(project, configuration);
+            ResolvedConfiguration resolvedConfiguration = SingleMessageLogger.whileDisabled(() -> configuration.getResolvedConfiguration());
```
Comment (yahavi) [inline build-info-extractor-gradle/src/main/groovy/org/jfrog/gradle/plugin/artifactory/extractor/GradleBuildInfoExtractor.java]:
```suggestion
            ResolvedConfiguration resolvedConfiguration = SingleMessageLogger.whileDisabled(configuration::getResolvedConfiguration);
```

**Comment by ghale** [comment]
I've signed the CLA.

**Comment by yahavi** [comment]
Thanks, @ghale. 
Did you notice my inline comment?

**Comment by ghale** [comment]
Actually, I didn't notice it before, but I've applied your suggestion.

**Comment by etiennestuder** [comment]
@yahavi Is there anything left for use from Gradle to do before this can get merged and released?

**Comment by yahavi** [comment]
@etiennestuder, no. Merged.
Thanks, @ghale!

**Comment by etiennestuder** [comment]
@yahavi Awesome :-) Thanks for the merge. This fix will unblock several of our customers. Do you have an ETA on a new release that includes these fixes?

**Comment by yahavi** [comment]
@etiennestuder, we originally planned to release this by the end of the month, but we'll make an effort to release it by the end of the week.

**Comment by etiennestuder** [comment]
@yahavi Oh wow. These are awesome news. Thanks very much in advance.

**Comment by yahavi** [comment]
@etiennestuder @ghale 
Gradle Artifactory plugin 4.12.0 is released.

**Comment by ogipogi** [comment]
Hi @yahavi 
Thank you very much for releasing 4.12.0.

I didn't find the version here:
https://github.com/jfrog/build-info/releases/tag/build-info-gradle-extractor-4.12.0

**Comment by etiennestuder** [comment]
Awesome, thanks very much @yahavi !

**Comment by yahavi** [comment]
Thanks @ogipogi
Try now

**Comment by ogipogi** [comment]
Thank you @yahavi 
It works.

**Pull request merged**